pkg/storage/mvcc_incremental_iterator.go

```diff
@@ -630,8 +627,12 @@ func (i *MVCCIncrementalIterator) NextIgnoringTime() {
 
 		// We have encountered an intent but it does not lie in the timestamp span
 		// (startTime, endTime] so we do not throw an error, and attempt to move to
-		// the next valid KV.
+		// the intent's corresponding provisional value.
+		//
+		// TODO(msbulter): investigate if it's clearer for the caller to emit the intent rather than the
+		// provisional value
 		if i.meta.Txn != nil && i.intentPolicy != MVCCIncrementalIterIntentPolicyEmit {
```

Comment:
@stevendanna should this policy surface intents that are outside of the time range? If so, we may want to update the docstring for this enum val. Currently it reads as:

```
	// MVCCIncrementalIterIntentPolicyEmit will return intents to
	// the caller if they are inside the time range. Intents
	// outside of the time range will be filtered without error.
	//
	// TODO(ssd): If we relaxed the requirement that intents are
	// filtered by time range, we could avoid parsing intents
	// inside the iterator and leave it to the caller to deal
	// with.
```

Reply:
I kinda think it should.  It just seems prone to error to emit the provisional value without emitting the intent before it.

Reply:
But, you could punt that to a follow up PR if you want. Or an "TODO(ssd)"

Reply:
Changed the docstring!

Reply:
Yeah, I'm with Steven on this.